src/commerce-handler.js

```diff
@@ -61,6 +116,182 @@ CommerceHandler.prototype.logCommerceEvent = function(event) {
             21: ProductRemoveFromWishlist,
             22: ProductImpression
         */
+    let events = [];
+
+    switch (event.EventCategory) {
+        case ProductActionTypes.Impression:
+            events = buildImpressionEvents(event);
+            break;
+        case PromotionType.PromotionClick:
+        case PromotionType.PromotionView:
+            events = buildPromotionEvents(event);
+            break;
+        default:
+            events = buildProductActionEvents(event);
+    };
+
+    events.forEach((event) => {
+        window.heap.track(event.Name, event.Properties);
+    });
+};
+
+function buildImpressionEvents(event) {
+    let events = [];
+
+    if (event.ProductImpressions.length > 0) {
+        event.ProductImpressions.forEach((impression) => {
+            let productSkus = [];
+            if (impression.ProductList.length > 0) {
+                impression.ProductList.forEach((product) => {
+                    let [productEvent, productSku] = buildProductEvent(product);
+                    events.push(productEvent);
+                    if(productSku) {
+                        productSkus.push(productSku);
+                    }
+                })
+            }
+
+            events.push(buildActionEvent(event, HeapConstants.EventNameImpression, productSkus))
+        })
+    }
+
+
+    return events;
 };
 
+function buildPromotionEvents(event) {
+    let events = [];
+    let promotionIds = [];
+    if (event.PromotionAction.PromotionList.length > 0) {
+        event.PromotionAction.PromotionList.forEach((promotion) => {
+            let [promotionEvent, promotionId] = buildPromotionEvent(promotion);
```

Comment:
Sadly, you can't destructure here because we don't have a transpiler to convert this to ES5.